share/dkg/pedersen/dkg.go

```diff
@@ -163,7 +163,7 @@ func NewDistKeyHandler(c *Config) (*DistKeyGenerator, error) {
 
 	var newThreshold int
 	if c.Threshold != 0 {
-		newThreshold = c.Threshold
+		newThreshold = int(c.Threshold)
```

Comment:
Why not changing the `newThreshold` type  above to match the expected `Threshold` type rather?